src/schema.ts

```diff
@@ -182,17 +199,80 @@ export class Schema {
 }
 
 /**
- * @return a summary with the correct distinct property given a max number of bins
+ * @return a summary of the binning scheme determined from the given max number of bins
  */
-function binSummary(maxbins: number, summary: Summary) {
-  const bin = dlBin({
+function binSummary(maxbins: number, summary: Summary): Summary {
+  const bin: Bin = dl.bins({
     min: summary.min,
     max: summary.max,
     maxbins: maxbins
   });
-  const binSum = extend({}, summary);
-  binSum.distinct = (bin.stop - bin.start) / bin.step;
-  return binSum;
+
+  return binStats(bin, summary);
+}
+
+function timeUnitSummary(unit: TimeUnit, summary: Summary, type: Type): Summary {
+  var bin: Bin;
+  switch (unit) {
+    // these are the units that dl.bins.date supports
+    case TimeUnit.YEAR:
+    case TimeUnit.MONTH:
+    case TimeUnit.DAY:
+    case TimeUnit.HOURS:
+    case TimeUnit.MINUTES:
+    case TimeUnit.SECONDS:
+      bin = dl.bins.date({
```

Comment:
Actually, the whole idea of using `dl.bins.date` is totally incorrect.  

I was curious, so I read and test it a bit

From the docs:

>  this method first determines an appropriate time unit (years, months, days, etc) and then uses dl.bin to produce a binning scheme.

It still perform binning, which means that it might not directly count # of values in between and might also adjust the start / stop.  I suggest that you add a number of test cases manually and see if the output is correct.  

I tried a few in the console of the datalib homepage and it doesn't seem correct. 

> var b = dl.bins.date({
  min: Date.parse('2000-01-01'),
  max: Date.parse('2005-12-31'), 
  unit: 'month' // equivalent to VL's `yearmonth`
}); 

I would except to get cardinality 5*12 = 60. However

> (b.stop - b.start)/b.step = 13
> (b.stop - b.start) = 78

Thus, neither of this is correct.